### DISCUSSION TITLE
I want to check MIDI note before it plays

### DISCUSSION THREAD

**Discussion by fapplin** [FortySevenEffects/arduino_midi_library#280] [Q&A] [answered]
I want to have a note played only when certain criteria is met. In the code below, I send the note to a function which will determine whether it should play or not (if I get a positive result back). The LED lights up correctly, but the note plays no matter what (I can even comment out the MIDI.sendNoteON()). What am I doing wrong? 

```c++
    if (MIDI.read())                // Is there a MIDI message incoming ?
    {
      switch (MIDI.getType())     // Get the type of the message we caught
      {
        case midi::NoteOn:  {
            chan = MIDI.getChannel();
            pitch = MIDI.getData1();
            velocity = MIDI.getData2();

            arr_pos = binary_search(scales, 127, pitch);

            if (arr_pos != -1)
            {
              digitalWrite(ledPin, HIGH);  // Any Note-Off turns on LED
              MIDI.sendNoteOn(arr_pos, velocity, chan);
            } //end if
          } //end case
          break;
     } //end switch
} //end if

```

**Comment by franky47** [comment, marked as answer]
Hum could it be that the SoftThru feature is replaying the incoming note?

You may want to disable it in `setup`:

```c++
void setup()
{
  MIDI.begin();
  MIDI.turnThruOff();
}
```

**Reply by fapplin** [reply]
Now, I get no sound at all.

**Reply by fapplin** [reply]
Got it working. Had to reset my synth.

**Reply by franky47** [reply]
Good to know!

FIY, the whole logic block of checking whether there's a message, switching over its type and getting its parameters can be omitted by using callbacks:
https://github.com/FortySevenEffects/arduino_midi_library/wiki/Using-Callbacks

**Reply by fapplin** [reply]
That's how I started, but when I kept get notes played no matter what - I thought I try the more primitive approach. I'll make a copy of what I have and try it with the callbacks, again (with the added MIDI.turnThruOff(). Thanks for the help.
